src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -84,6 +90,34 @@ internal partial class CSharpSelectionValidator : SelectionValidator
                 cancellationToken).ConfigureAwait(false);
         }
 
+        private bool SpanContainsLocalFunction(TextSpan finalSpan, SemanticModel model, SyntaxNode root)
+        {
+            var nodes = root.DescendantNodes().Where(n => finalSpan.Contains(n.Span));
+            foreach (var node in nodes)
+            {
+                if (node.IsKind(SyntaxKind.LocalFunctionStatement))
+                {
+                    return true;
+                }
+
+                if (node is IdentifierNameSyntax id)
+                {
+                    var symbolInfo = model.GetSymbolInfo(id);
+                    if (symbolInfo.Symbol is IMethodSymbol method &&
+                        method.MethodKind == MethodKind.LocalFunction)
+                    {
+                        return true;
+                    }
+                }
+
+                if (node.HasAncestor<LocalFunctionStatementSyntax>())
```

Comment:
No -- my tests expect that you can't. This is also unsafe because values may be captured inside the local function and data flow analysis will not reflect that.